test/gtest/ucs/test_conn_match.cc

```diff
@@ -0,0 +1,251 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#include <common/test.h>
+
+#include <ucs/datastruct/conn_match.h>
+#include <ucs/sys/sys.h>
+
+#include <vector>
+
+class test_conn_match : public ucs::test {
+public:
+    typedef struct {
+        ucs_conn_match_elem_t       conn_match;
+        ucs_conn_match_queue_type_t queue_type;
+        void                        *dest_address;
+        ucs_conn_sn_t               conn_sn;
+    } conn_elem_t;
+
+protected:
+    static const void* get_address(const ucs_conn_match_elem_t *conn_match) {
+        return ucs_container_of(conn_match, conn_elem_t,
+                                conn_match)->dest_address;
+    }
+
+    static ucs_conn_sn_t get_conn_sn(const ucs_conn_match_elem_t *conn_match) {
+        return ucs_container_of(conn_match, conn_elem_t,
+                                conn_match)->conn_sn;
+    }
+
+    static const char *address_str(const void *address, char *str,
+                                   size_t max_size) {
+        return ucs_strncpy_safe(str, static_cast<const char*>(address),
+                                ucs_min(m_conn_match_ctx.address_length,
+                                        max_size));
+    }
+
+    void conn_match_init(size_t address_length) {
```

Comment:
move this to "virtual void init()"

Reply:
it couldn't be moved, since we need init conn matching by length

Reply:
i think we can have predefined length like 20bytes for example

Reply:
done